task.sql

```diff
@@ -1,11 +1,56 @@
--- Use our database
-USE ShopDB; 
+DROP DATABASE IF EXISTS ShopDB;
+CREATE DATABASE ShopDB;
+USE ShopDB;
 
--- Some data should be created outside the transaction (here)
+CREATE TABLE Products (
+                          ID INT AUTO_INCREMENT,
+                          Name VARCHAR(50),
+                          Description VARCHAR(100),
+                          Price INT,
+                          WarehouseAmount INT,
+                          PRIMARY KEY (ID)
+);
 
--- Start the transaction 
-START TRANSACTION; 
+CREATE TABLE Customers (
+                           ID INT AUTO_INCREMENT,
+                           FirstName VARCHAR(50),
+                           LastName VARCHAR(50),
+                           Email VARCHAR(50),
+                           Address VARCHAR(50),
+                           PRIMARY KEY (ID)
+);
+
+CREATE TABLE Orders (
+                        ID INT AUTO_INCREMENT,
+                        CustomerID INT,
+                        FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL,
+                        Date DATE,
+                        PRIMARY KEY (ID)
+);
+
+CREATE TABLE OrderItems (
+                            ID INT AUTO_INCREMENT,
+                            OrderID INT,
+                            FOREIGN KEY (OrderID) REFERENCES Orders(ID) ON DELETE SET NULL,
+                            ProductID INT,
+                            Count INT,
+                            FOREIGN KEY (ProductID) REFERENCES Products(ID) ON DELETE SET NULL,
```

Comment:
The foreign key constraint on `ProductID` with `ON DELETE SET NULL` may lead to `NULL` values in `ProductID` if a product is deleted. Consider using `ON DELETE CASCADE` if you want to delete order items associated with a deleted product, or ensure your application logic handles `NULL` values appropriately.